packages/shared/src/sdk-versioning/sdk-payload.ts

```diff
@@ -218,10 +222,217 @@ export const scrubSavedGroups = (
   return savedGroupsValues;
 };
 
+// Maximum depth for recursive saved group resolution
+const MAX_SAVED_GROUP_DEPTH = 10;
+
+export const SAVED_GROUP_ERROR_MAX_DEPTH = "__sgMaxDepth__";
+export const SAVED_GROUP_ERROR_CYCLE = "__sgCycle__";
+export const SAVED_GROUP_ERROR_INVALID = "__sgInvalid__";
+export const SAVED_GROUP_ERROR_UNKNOWN = "__sgUnknown__";
+
+export function conditionHasSavedGroupErrors(
+  condition: unknown,
+  ignoreCycleErrors: boolean = false,
+) {
+  if (!condition) return false;
+
+  const src =
+    typeof condition === "object"
+      ? JSON.stringify(condition)
+      : String(condition);
+
+  const errorMarkers = [
+    SAVED_GROUP_ERROR_INVALID,
+    SAVED_GROUP_ERROR_UNKNOWN,
```

Comment:
Does the importer work with this?  Wouldn't the dummy `__unknown__` saved group ids cause an `ERROR_UNKNOWN` error?

Reply:
it worked, but not for the correct reason. the 2-pass system masked the failed state. will fix the validator bypass logic